packages/yew/src/html/classes.rs

```diff
@@ -153,7 +178,18 @@ impl From<&'static str> for Classes {
 
 impl From<String> for Classes {
     fn from(t: String) -> Self {
-        Self::from(&t)
+        match t.contains(|c: char| c.is_whitespace()) && !t.is_empty() {
+            // If the string only contains a single class, we can just use it
+            // directly (rather than cloning it into a new string). Need to make
+            // sure it's not empty, though.
+            false => match t.is_empty() {
+                true => Self::new(),
+                false => Self {
+                    set: IndexSet::from_iter([Cow::Owned(t)]),
+                },
+            },
+            true => Self::from(&t),
+        }
```

Comment:
```suggestion
        if t.is_empty() {
            return Self::new();
        }
        match t.contains(|c: char| c.is_whitespace()) {
            // If the string only contains a single class, we can just use it
            // directly (rather than cloning it into a new string). Need to make
            // sure it's not empty, though.
            false => Self {
                set: IndexSet::from_iter([Cow::Owned(t)]),
            },
            true => Self::from(&t),
        }
```

`t.is_empty()` only needs to be checked once.

Reply:
Oh good catch